sound/soc/sof/ipc4-topology.c

```diff
@@ -1485,6 +1485,85 @@ static int sof_ipc4_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget
 	return ret;
 }
 
+static int sof_ipc4_get_queue_id(struct snd_sof_widget *swidget, const char *peer_name,
+				  bool is_src_widget)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct ida *queue_ida;
+	char *bind_widget_name;
+	char *pin_binding;
+	int num_pins;
+	int i;
+
+	/* For source widget in a route, we should check its sink pin binding array and
+	 * sink_queue_ida, for sink widget in a route, we should check its source pin bingding
+	 * array and source_queue_ida.
+	 */
+	if (is_src_widget) {
+		pin_binding = swidget->sink_pin_binding;
+		queue_ida = &swidget->sink_queue_ida;
+		num_pins = swidget->num_sink_pins;
+	} else {
+		pin_binding = swidget->src_pin_binding;
+		queue_ida = &swidget->src_queue_ida;
+		num_pins = swidget->num_source_pins;
+	}
+
+	if (num_pins < 1) {
```

Comment:
I think you should move this chck to the first patch where you parse the number of pins